packages/nitro-protocol/contracts/ninja-nitro/readme.md

```diff
@@ -0,0 +1,97 @@
+# Ninja Nitro
+
+Ninja-Nitro is an alternative implementation of Nitro protocol, with a different contract architecture to the main implementation (Mono-Nitro) in this package. The additional files should all have namespaced paths.
```

Comment:
This may seem pedantic, but I would prefer to use `Mono-Nitro` to refer to a monolithic nitro architecture, one with _one_ smart contract. The existing architecture decouples adjudication from funding across multiple contracts.

Reply:
The existing architecture is monolithic in the sense of the total number of adjudicators (one), which is the important distinction with ninja-nitro. 

I agree that there is a potentially interesting architecture with one contract _total_. It's more monolithic, or monolithic on more ways than the existing architecture. 

I'd still like to use Mono-Nitro here, since it is immediately useful and will continue to be so in the short term. I would view the More-Mono-Nitro as being a replacement for the current architecture, rather than an alternative (in the way that Ninja-Nitro is an alternative). So it could still be called Mono-Nitro?

That said, I'd be willing to swap this term out for another one if you have a good suggestion? Or I suppose I could just use "the existing implementation" everywhere...

Reply:
> I would view the More-Mono-Nitro as being a replacement for the current architecture, rather than an alternative (in the way that Ninja-Nitro is an alternative). So it could still be called Mono-Nitro?

There is a third architecture for L1 channels which has:
1. a single AssetHolder
2. a single adjudicator

Off the top of my head, I might call this a "duplex architecture", "Duo Nitro", or something that emphasizes that adjudication and asset management are decoupled.

I think the difference between this and the current architecture is closer to an implementation detail, as I cannot think of many differences in what on-chain operations they'd enable. In particular, they both enable the following for L1 channels:
- AssetHolders live on L1
- the Adjudicator lives on an L2.
This allows peers in an L1 channel to dispute in L2 for much lower fees.

The "mono mono nitro" might be more suitable for satisfying some (not-yet-understood) hub feature features as well as minimizing the number of storage slots required (we need at least one, and I think we can do that with "mono mono nitro".

Reply:
I'll just use Vanilla Nitro for now 095a3b6 . It doesn't need to be the long term name for this.